cpp/src/arrow/util/CMakeLists.txt

```diff
@@ -54,7 +54,7 @@ endif()
 
 ADD_ARROW_TEST(bit-util-test)
 ADD_ARROW_TEST(checked-cast-test)
-ADD_ARROW_TEST(compression-test)
+ADD_ARROW_TEST(compression-test EXTRA_LINK_LIBS zlib)
```

Comment:
Seems like this should transitively linked as a result of linking to libarrow.a. We should make a follow up to make sure that CMake is set to include transitive dependencies when statically linking

Reply:
If we add `zlib` to `ARROW_STATIC_LINK_LIBS` even when we use zlib as shared library, we can remove this: https://github.com/apache/arrow/pull/2483/commits/db17bdb4c00ee38f0ea09fe7e4fc881290a13e5a

I'm not sure that it's a good manner in this project...

Reply:
OK